### ISSUE TITLE
Improve how batch action works with other plugins

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#3226]
Batch actions is a commonly used plugin but when used in combination with certain other plugins/hooks, we see some unexpected or undesired behavior.

Currently, when adding [batch actions to clickable rows](https://pages.github.ibm.com/cdai-design/pal/components/data-table/clickable-row-items/usage/#interactions), the active state (blue bar) is not visible. Based on the design guidance, in this context, only selecting the checkbox should select the row, while clicking on the whole row should change the active row but not necessarily affect selection. 
This could be confusing so probably best to create an example and review with our designers. This might need to be a separate issue depending how easily we’re able to get this working.

```[tasklist]
### Tasks
- [ ] https://github.com/carbon-design-system/ibm-products/issues/3334
- [x] Add examples for clickable rows with batch actions
```
